src/creates/createIssue.ts

```diff
@@ -197,11 +210,30 @@ export const createIssue = {
         dynamic: "team.id.name",
         altersDynamicFields: true,
       },
-      {
-        required: true,
-        label: "Title",
-        helpText: "The title of the issue",
-        key: "title",
+      // Populate title with the template's title if applicable
+      async function (z: ZObject, bundle: Bundle) {
+        const baseTitle = {
+          label: "Title",
+          helpText: "The title of the issue",
+          key: "title",
+          required: true,
+        };
+        if (bundle.inputData.templateId) {
+          const query = `
+            query ZapierTemplate($templateId: String!) {
+              template(id: $templateId) {
+                templateData
+              }
+            }`;
+          const response = await fetchFromLinear(z, bundle, query, { templateId: bundle.inputData.templateId });
+
+          const template = response.json.data.template;
+          const templateData = JSON.parse(template.templateData);
+          if (templateData.title) {
```

Comment:
To confirm – the template issue title would be used even without this logic, this is just a nicety to display it in Zapier? Or is there more to it.

Reply:
It’s also because we mark title as a required field here

Reply:
I suppose the downside is if the template title changes inside of Linear the Zapier will be stuck with the old title.

Anyway to make title optional based on existence of another param?

Reply:
Let me see

Reply:
Got it to work

Reply:
lovely